### ISSUE TITLE
[disk hot unplug] removing of  attachment pod if volume is present in vmi.status.volumeStatus but removed from vmi.spec.volumes

### ISSUE THREAD

**Issue by t33m** [kubevirt/kubevirt#13481]
**What happened**:
Hi folks,

I have a question about disk unplug scenario. I see that attachment pod is removing if the VMI has no hotplugVolumes in the spec https://github.com/kubevirt/kubevirt/blob/47d3895beb5b8617f73be9c5daac04f5dcfb4c61/pkg/virt-controller/watch/vmi/vmi.go#L1071-L1071

but when we call removeVolume on hotplugged volume via virtctl, the volume is removed from the VMI spec but available in status.volumeStatus. And it seems that virt-controller removes attachment pod and doesn't wait for volume to be successfully detached from qemu via libvirt by call from virt-handler https://github.com/kubevirt/kubevirt/blob/47d3895beb5b8617f73be9c5daac04f5dcfb4c61/pkg/virt-controller/watch/vmi/vmi.go#L1672-L1672.

Could you confirm or deny my guess?

**Environment**:
- KubeVirt version (use `virtctl version`): main

**Comment by t33m** [comment]
@awels , @aburdenthehand , what do you think about my assumption?

**Comment by aburdenthehand** [comment]
Sorry for missing this @t33m. I have no idea but I'll see if I can get @awels eyes on it.

**Comment by awels** [comment]
Yes this is correct, but it is intentional. We know eventually libvirt will remove the hotplugged volume from the VM and release the lock on the file/device. So we can actually delete the attachment pod if there are no volumes left. If there are other hotplugged volumes we create a new attachment pod that only contains those volumes.

**Comment by mhenriks** [comment]
@t33m https://github.com/kubevirt/kubevirt/pull/14998 should fix this
